[TEST-FAILURE] MigrationAwareServiceTest.testPartitionDataSize_whenNodeGracefullyShutdown

```
java.lang.AssertionError: expected:<271> but was:<215>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.spi.MigrationAwareServiceTest$3.run(MigrationAwareServiceTest.java:222)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:825)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:839)
	at com.hazelcast.spi.MigrationAwareServiceTest.assertSize(MigrationAwareServiceTest.java:211)
	at com.hazelcast.spi.MigrationAwareServiceTest.testPartitionDataSize_whenNodeGracefullyShutdown(MigrationAwareServiceTest.java:160)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-IbmJDK1.7/com.hazelcast$hazelcast/756/testReport/junit/com.hazelcast.spi/MigrationAwareServiceTest/testPartitionDataSize_whenNodeGracefullyShutdown/

--- a/hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterJoinManager.java
+++ b/hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterJoinManager.java
@@ -36,6 +36,7 @@ import com.hazelcast.logging.ILogger;
 import com.hazelcast.nio.Address;
 import com.hazelcast.nio.Connection;
 import com.hazelcast.nio.Packet;
+import com.hazelcast.partition.PartitionRuntimeState;
 import com.hazelcast.security.Credentials;
 import com.hazelcast.spi.Operation;
 import com.hazelcast.spi.OperationService;
@@ -440,8 +441,9 @@ public class ClusterJoinManager {
                 Operation[] postJoinOps = nodeEngine.getPostJoinOperations();
                 boolean isPostJoinOperation = postJoinOps != null && postJoinOps.length > 0;
                 PostJoinOperation postJoinOp = isPostJoinOperation ? new PostJoinOperation(postJoinOps) : null;
-                Operation operation = new FinalizeJoinOperation(createMemberInfoList(clusterService.getMemberImpls()), postJoinOp,
-                        clusterClock.getClusterTime(), clusterStateManager.getState(), false);
+                Operation operation = new FinalizeJoinOperation(createMemberInfoList(clusterService.getMemberImpls()),
+                        postJoinOp, clusterClock.getClusterTime(), clusterStateManager.getState(),
+                        node.partitionService.createPartitionState(), false);
                 nodeEngine.getOperationService().send(operation, target);
             } else {
                 sendMasterAnswer(target);
@@ -492,10 +494,12 @@ public class ClusterJoinManager {
 
                 int count = members.size() - 1 + setJoins.size();
                 List<Future> calls = new ArrayList<Future>(count);
+                PartitionRuntimeState partitionState = node.partitionService.createPartitionState();
                 for (MemberInfo member : setJoins) {
                     long startTime = clusterClock.getClusterStartTime();
                     Operation joinOperation = new FinalizeJoinOperation(memberInfos, postJoinOp, time,
-                            clusterService.getClusterId(), startTime, clusterStateManager.getState());
+                            clusterService.getClusterId(), startTime,
+                            clusterStateManager.getState(), partitionState);
                     calls.add(invokeClusterOperation(joinOperation, member.getAddress()));
                 }
                 for (MemberImpl member : members) {

--- a/hazelcast/src/main/java/com/hazelcast/cluster/impl/operations/FinalizeJoinOperation.java
+++ b/hazelcast/src/main/java/com/hazelcast/cluster/impl/operations/FinalizeJoinOperation.java
@@ -20,13 +20,16 @@ import com.hazelcast.cluster.ClusterState;
 import com.hazelcast.cluster.MemberInfo;
 import com.hazelcast.cluster.impl.ClusterServiceImpl;
 import com.hazelcast.core.Member;
+import com.hazelcast.instance.Node;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.partition.PartitionRuntimeState;
 import com.hazelcast.spi.Operation;
 import com.hazelcast.spi.OperationAccessor;
 import com.hazelcast.spi.OperationService;
 import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationservice.InternalOperationService;
 
 import java.io.IOException;
 import java.util.Collection;
@@ -39,6 +42,7 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
     public static final int FINALIZE_JOIN_MAX_TIMEOUT = 60;
 
     private PostJoinOperation postJoinOp;
+    private PartitionRuntimeState partitionRuntimeState;
     private String clusterId;
     private long clusterStartTime;
     private ClusterState clusterState;
@@ -46,20 +50,23 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
     public FinalizeJoinOperation() {
     }
 
-    public FinalizeJoinOperation(Collection<MemberInfo> members, PostJoinOperation postJoinOp,
-                                 long masterTime, String clusterId, long clusterStartTime, ClusterState clusterState) {
+    public FinalizeJoinOperation(Collection<MemberInfo> members, PostJoinOperation postJoinOp, long masterTime,
+            String clusterId, long clusterStartTime, ClusterState clusterState,
+            PartitionRuntimeState partitionRuntimeState) {
         super(members, masterTime, true);
         this.postJoinOp = postJoinOp;
         this.clusterId = clusterId;
-        this.clusterStartTime  = clusterStartTime;
+        this.clusterStartTime = clusterStartTime;
         this.clusterState = clusterState;
+        this.partitionRuntimeState = partitionRuntimeState;
     }
 
-    public FinalizeJoinOperation(Collection<MemberInfo> members, PostJoinOperation postJoinOp,
-                                 long masterTime, ClusterState clusterState, boolean sendResponse) {
+    public FinalizeJoinOperation(Collection<MemberInfo> members, PostJoinOperation postJoinOp, long masterTime,
+            ClusterState clusterState, PartitionRuntimeState partitionRuntimeState, boolean sendResponse) {
         super(members, masterTime, sendResponse);
         this.postJoinOp = postJoinOp;
         this.clusterState = clusterState;
+        this.partitionRuntimeState = partitionRuntimeState;
     }
 
     @Override
@@ -79,11 +86,53 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
             return;
         }
 
+        initClusterStates(clusterService);
+
+        processMemberUpdate();
+
+        processPartitionState();
+
+        sendPostJoinOperations();
+
+        runPostJoinOp();
+    }
+
+    private void initClusterStates(ClusterServiceImpl clusterService) {
         clusterService.initialClusterState(clusterState);
         clusterService.setClusterId(clusterId);
         clusterService.getClusterClock().setClusterStartTime(clusterStartTime);
+    }
 
-        processMemberUpdate();
+    private void processPartitionState() {
+        if (partitionRuntimeState == null) {
+            return;
+        }
+
+        partitionRuntimeState.setEndpoint(getCallerAddress());
+        ClusterServiceImpl clusterService = getService();
+        Node node = clusterService.getNodeEngine().getNode();
+        node.partitionService.processPartitionRuntimeState(partitionRuntimeState);
+    }
+
+    private void runPostJoinOp() {
+        if (postJoinOp == null) {
+            return;
+        }
+
+        ClusterServiceImpl clusterService = getService();
+        NodeEngineImpl nodeEngine = clusterService.getNodeEngine();
+        InternalOperationService operationService = nodeEngine.getOperationService();
+
+        postJoinOp.setNodeEngine(nodeEngine);
+        OperationAccessor.setCallerAddress(postJoinOp, getCallerAddress());
+        OperationAccessor.setConnection(postJoinOp, getConnection());
+        postJoinOp.setOperationResponseHandler(createEmptyResponseHandler());
+        operationService.runOperationOnCallingThread(postJoinOp);
+    }
+
+    private void sendPostJoinOperations() {
+        final ClusterServiceImpl clusterService = getService();
+        final NodeEngineImpl nodeEngine = clusterService.getNodeEngine();
 
         // Post join operations must be lock free; means no locks at all;
         // no partition locks, no key-based locks, no service level locks!
@@ -100,14 +149,6 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
                 }
             }
         }
-
-        if (postJoinOp != null) {
-            postJoinOp.setNodeEngine(nodeEngine);
-            OperationAccessor.setCallerAddress(postJoinOp, getCallerAddress());
-            OperationAccessor.setConnection(postJoinOp, getConnection());
-            postJoinOp.setOperationResponseHandler(createEmptyResponseHandler());
-            operationService.runOperationOnCallingThread(postJoinOp);
-        }
     }
 
     @Override
@@ -121,6 +162,7 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
         out.writeUTF(clusterId);
         out.writeLong(clusterStartTime);
         out.writeUTF(clusterState.toString());
+        out.writeObject(partitionRuntimeState);
     }
 
     @Override
@@ -135,6 +177,7 @@ public class FinalizeJoinOperation extends MemberInfoUpdateOperation implements 
         clusterStartTime = in.readLong();
         String stateName = in.readUTF();
         clusterState = ClusterState.valueOf(stateName);
+        partitionRuntimeState = in.readObject();
     }
 
     @Override

--- a/hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java
+++ b/hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java
@@ -97,6 +97,11 @@ public class DefaultNodeExtension implements NodeExtension {
     }
 
     @Override
+    public boolean isStartCompleted() {
+        return node.joined();
+    }
+
+    @Override
     public SecurityContext getSecurityContext() {
         logger.warning("Security features are only available on Hazelcast Enterprise!");
         return null;

--- a/hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java
+++ b/hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java
@@ -57,6 +57,12 @@ public interface NodeExtension {
     void afterStart();
 
     /**
+     * Returns true if the instance has started
+     * @return true if the instance has started
+     */
+    boolean isStartCompleted();
+
+    /**
      * Creates a <tt>SerializationService</tt> instance to be used by this <tt>Node</tt>.
      *
      * @return a <tt>SerializationService</tt> instance

--- a/hazelcast/src/main/java/com/hazelcast/partition/PartitionDataSerializerHook.java
+++ b/hazelcast/src/main/java/com/hazelcast/partition/PartitionDataSerializerHook.java
@@ -29,6 +29,7 @@ public final class PartitionDataSerializerHook implements DataSerializerHook {
 
     public static final int F_ID = FactoryIdHelper.getFactoryId(PARTITION_DS_FACTORY, PARTITION_DS_FACTORY_ID);
 
+    public static final int PARTITION_STATE = 1;
     public static final int PARTITIONS = 2;
 
     @Override
@@ -42,6 +43,8 @@ public final class PartitionDataSerializerHook implements DataSerializerHook {
             @Override
             public IdentifiedDataSerializable create(int typeId) {
                 switch (typeId) {
+                    case PARTITION_STATE:
+                        return new PartitionRuntimeState();
                     case PARTITIONS:
                         return new PartitionsResponse();
                     default:

--- a/hazelcast/src/main/java/com/hazelcast/partition/PartitionRuntimeState.java
+++ b/hazelcast/src/main/java/com/hazelcast/partition/PartitionRuntimeState.java
@@ -22,6 +22,7 @@ import com.hazelcast.nio.Address;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 
 import java.io.IOException;
 import java.util.ArrayList;
@@ -32,10 +33,9 @@ import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 
-public class PartitionRuntimeState implements DataSerializable {
+public final class PartitionRuntimeState implements IdentifiedDataSerializable {
 
-    protected final ArrayList<MemberInfo> members = new ArrayList<MemberInfo>(100);
-
+    private final List<MemberInfo> members = new ArrayList<MemberInfo>(100);
     private final Collection<ShortPartitionInfo> partitionInfos = new LinkedList<ShortPartitionInfo>();
     private ILogger logger;
     private int version;
@@ -62,12 +62,12 @@ public class PartitionRuntimeState implements DataSerializable {
         completedMigrations = migrationInfos != null ? migrationInfos : new ArrayList<MigrationInfo>(0);
     }
 
-    protected void addMemberInfo(MemberInfo memberInfo, Map<Address, Integer> addressIndexes, int memberIndex) {
+    private void addMemberInfo(MemberInfo memberInfo, Map<Address, Integer> addressIndexes, int memberIndex) {
         members.add(memberIndex, memberInfo);
         addressIndexes.put(memberInfo.getAddress(), memberIndex);
     }
 
-    protected void setPartitions(InternalPartition[] partitions, Map<Address, Integer> addressIndexes) {
+    private void setPartitions(InternalPartition[] partitions, Map<Address, Integer> addressIndexes) {
         List<String> unmatchAddresses = new LinkedList<String>();
         for (InternalPartition partition : partitions) {
             ShortPartitionInfo partitionInfo = new ShortPartitionInfo(partition.getPartitionId());
@@ -243,5 +243,15 @@ public class PartitionRuntimeState implements DataSerializable {
                 addressIndexes[i] = in.readInt();
             }
         }
+    }
+
+    @Override
+    public int getFactoryId() {
+        return PartitionDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getId() {
+        return PartitionDataSerializerHook.PARTITION_STATE;
     }
 }

--- a/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java
+++ b/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java
@@ -492,8 +492,7 @@ public class InternalPartitionServiceImpl implements InternalPartitionService, M
                     }
 
                     // send initial partition table to newly joined node.
-                    Collection<MemberImpl> members = getCurrentMembersAndMembersRemovedWhileNotClusterNotActive();
-                    PartitionStateOperation op = new PartitionStateOperation(createPartitionState(members));
+                    PartitionStateOperation op = new PartitionStateOperation(createPartitionState());
                     nodeEngine.getOperationService().send(op, member.getAddress());
                 }
             } finally {
@@ -608,7 +607,15 @@ public class InternalPartitionServiceImpl implements InternalPartitionService, M
         }
     }
 
+    public PartitionRuntimeState createPartitionState() {
+        return createPartitionState(getCurrentMembersAndMembersRemovedWhileNotClusterNotActive());
+    }
+
     private PartitionRuntimeState createPartitionState(Collection<MemberImpl> members) {
+        if (!initialized) {
+            return null;
+        }
+
         lock.lock();
         try {
             List<MemberInfo> memberInfos = new ArrayList<MemberInfo>(members.size());
@@ -707,7 +714,7 @@ public class InternalPartitionServiceImpl implements InternalPartitionService, M
         return calls;
     }
 
-    void processPartitionRuntimeState(PartitionRuntimeState partitionState) {
+    public void processPartitionRuntimeState(PartitionRuntimeState partitionState) {
         lock.lock();
         try {
             final Address sender = partitionState.getEndpoint();

--- a/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java
+++ b/hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java
@@ -486,7 +486,10 @@ public class InternalPartitionServiceImpl implements InternalPartitionService, M
             try {
                 migrationQueue.clear();
                 if (initialized) {
-                    migrationQueue.add(new RepartitioningTask());
+                    final ClusterState clusterState = nodeEngine.getClusterService().getClusterState();
+                    if (clusterState == ClusterState.ACTIVE) {
+                        migrationQueue.add(new RepartitioningTask());
+                    }
 
                     // send initial partition table to newly joined node.
                     Collection<MemberImpl> members = getCurrentMembersAndMembersRemovedWhileNotClusterNotActive();
@@ -631,7 +634,7 @@ public class InternalPartitionServiceImpl implements InternalPartitionService, M
             return;
         }
 
-        if (!isMigrationAllowed()) {
+        if (!isReplicaSyncAllowed()) {
             // migration is disabled because of a member leave, wait till enabled!
             return;
         }

--- a/hazelcast/src/main/java/com/hazelcast/partition/impl/PartitionStateOperation.java
+++ b/hazelcast/src/main/java/com/hazelcast/partition/impl/PartitionStateOperation.java
@@ -17,7 +17,7 @@
 package com.hazelcast.partition.impl;
 
 import com.hazelcast.cluster.impl.operations.JoinOperation;
-import com.hazelcast.instance.NodeState;
+import com.hazelcast.instance.Node;
 import com.hazelcast.nio.ObjectDataInput;
 import com.hazelcast.nio.ObjectDataOutput;
 import com.hazelcast.partition.InternalPartitionService;
@@ -48,8 +48,8 @@ public final class PartitionStateOperation extends AbstractOperation
 
     @Override
     public void run() {
-        NodeEngineImpl nodeEngine = (NodeEngineImpl) getNodeEngine();
-        if (nodeEngine.getNode().getState() == NodeState.PASSIVE) {
+        final Node node = ((NodeEngineImpl) getNodeEngine()).getNode();
+        if (!node.getNodeExtension().isStartCompleted()) {
             getLogger().warning("Partition table received before startup is completed. Caller: " + getCallerAddress());
             return;
         }

--- a/hazelcast/src/test/java/com/hazelcast/cluster/impl/BasicClusterStateTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cluster/impl/BasicClusterStateTest.java
@@ -19,6 +19,7 @@ package com.hazelcast.cluster.impl;
 import com.hazelcast.cluster.ClusterState;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
 import com.hazelcast.instance.GroupProperties;
 import com.hazelcast.instance.Node;
 import com.hazelcast.instance.NodeState;
@@ -43,7 +44,8 @@ import static org.junit.Assert.fail;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
-public class BasicClusterStateTest extends HazelcastTestSupport {
+public class BasicClusterStateTest
+        extends HazelcastTestSupport {
 
     @Test
     public void clusterState_isActive_whenInstancesStarted() {
@@ -318,8 +320,8 @@ public class BasicClusterStateTest extends HazelcastTestSupport {
     @Test(expected = IllegalArgumentException.class)
     public void changeClusterState_transaction_mustBe_TWO_PHASE() {
         HazelcastInstance hz = createHazelcastInstance();
-        hz.getCluster().changeClusterState(ClusterState.FROZEN,
-                new TransactionOptions().setTransactionType(TransactionType.LOCAL));
+        hz.getCluster()
+          .changeClusterState(ClusterState.FROZEN, new TransactionOptions().setTransactionType(TransactionType.LOCAL));
     }
 
     @Test
@@ -332,6 +334,41 @@ public class BasicClusterStateTest extends HazelcastTestSupport {
         Map map = hz.getMap(randomMapName());
         changeClusterStateEventually(hz, ClusterState.PASSIVE);
         map.get(1);
+    }
+
+    @Test(timeout = 60000)
+    public void test_noMigration_whenNodeLeaves_onClusterState_FROZEN() {
+        testNoMigrationWhenNodeLeaves(ClusterState.FROZEN);
+    }
+
+    @Test(timeout = 60000)
+    public void test_noMigration_whenNodeLeaves_onClusterState_PASSIVE() {
+        testNoMigrationWhenNodeLeaves(ClusterState.PASSIVE);
+    }
+
+    private void testNoMigrationWhenNodeLeaves(final ClusterState clusterState) {
+        final TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(3);
+        final HazelcastInstance master = factory.newHazelcastInstance();
+        HazelcastInstance other = factory.newHazelcastInstance();
+
+        final IMap<Object, Object> map = master.getMap(randomMapName());
+        for (int i = 0; i < 10000; i++) {
+            map.put(i, i);
+        }
+
+        changeClusterStateEventually(master, clusterState);
+
+        final Address otherAddress = getAddress(other);
+
+        other.shutdown();
+        assertClusterSizeEventually(1, master);
+
+        other = factory.newHazelcastInstance(otherAddress);
+
+        assertClusterSizeEventually(2, master);
+        assertClusterSizeEventually(2, other);
+
+        other.shutdown();
     }
 
     private static void assertClusterState(ClusterState expectedState, HazelcastInstance... instances) {

--- a/hazelcast/src/test/java/com/hazelcast/partition/impl/FrozenPartitionTableTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/partition/impl/FrozenPartitionTableTest.java
@@ -4,11 +4,13 @@ import com.hazelcast.cluster.ClusterState;
 import com.hazelcast.config.Config;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.nio.Address;
+import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.TestHazelcastInstanceFactory;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.test.annotation.Repeat;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
@@ -28,7 +30,7 @@ import static org.junit.Assert.assertNotNull;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
-public class FrozenPartitionTableTest  extends HazelcastTestSupport {
+public class FrozenPartitionTableTest extends HazelcastTestSupport {
 
     @Test
     public void partitionTable_isFrozen_whenNodesLeave_duringClusterStateIsFrozen() {
@@ -40,6 +42,7 @@ public class FrozenPartitionTableTest  extends HazelcastTestSupport {
         testPartitionTableIsFrozenDuring(ClusterState.PASSIVE);
     }
 
+    @Repeat(100)
     @Test
     public void partitionTable_isFrozen_whenMemberReJoins_duringClusterStateIsFrozen() {
         Config config = new Config();
@@ -63,7 +66,14 @@ public class FrozenPartitionTableTest  extends HazelcastTestSupport {
         assertClusterSizeEventually(3, hz3);
 
         for (HazelcastInstance instance : Arrays.asList(hz1, hz2, hz3)) {
-            assertPartitionTablesSame(partitionTable, getPartitionTable(instance));
+            final HazelcastInstance hz = instance;
+            assertTrueEventually(new AssertTask() {
+                @Override
+                public void run()
+                        throws Exception {
+                    assertPartitionTablesSame(partitionTable, getPartitionTable(hz));
+                }
+            });
         }
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/partition/impl/FrozenPartitionTableTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/partition/impl/FrozenPartitionTableTest.java
@@ -10,7 +10,6 @@ import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.TestHazelcastInstanceFactory;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
-import com.hazelcast.test.annotation.Repeat;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
@@ -42,7 +41,6 @@ public class FrozenPartitionTableTest extends HazelcastTestSupport {
         testPartitionTableIsFrozenDuring(ClusterState.PASSIVE);
     }
 
-    @Repeat(100)
     @Test
     public void partitionTable_isFrozen_whenMemberReJoins_duringClusterStateIsFrozen() {
         Config config = new Config();

